app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java

```diff
@@ -330,7 +330,16 @@ public String findThumbnailByFilename(String filename) throws IOException {
     }
 
     @Override
-    public String parseWikicode(String source) throws IOException {
+    public String readFeedback(String source, String wikidataQID) throws IOException {
```

Comment:
Can you use the JSON API instead of the XML one? You can add the same API in `OkHttpJsonApiClient` passing `.param("format", "json")` as an additional parameter.

Reply:
@maskaravivek when using OkHttpClient I am not getting the correct response:
I got 
```
{"batchcomplete":"","query":{"pages":{"-1":{"ns":1,"title":"Talk:Q11311478","missing":""}}}}
``` 
for the code:
```
HttpUrl.Builder urlBuilder = HttpUrl
                .parse(commonsBaseUrl)
                .newBuilder()
                .addQueryParameter("action", "query")
                .addQueryParameter("prop", "revisions")
                .addQueryParameter("rvprop", "content")
                .addQueryParameter("rvslots", "main")
                .addQueryParameter("titles", "Talk:Q11311478")
                .addQueryParameter("format", "json");

        Request request = new Request.Builder()
                .url(urlBuilder.build())
                .build();
        Response response = okHttpClient.newCall(request).execute();
        if (response.body() != null && response.isSuccessful()) {
            String json = response.body().string();
            MwQueryResponse mwQueryPage = gson.fromJson(json, MwQueryResponse.class);
            mwQueryPage.query().firstPage().revisions().get(0);
        }
```